packages/shared/src/hooks/useOnboardingChecklist.tsx

```diff
@@ -0,0 +1,156 @@
+      }),
+      createChecklistStep({
+        type: ActionType.VotePost,
+        step: {
+          title: 'Upvote or downvote a post',
+          description:
+            'Engage with the community by voting on posts based on their relevance and quality.',
+        },
+        actions,
+      }),
+      createChecklistStep({
+        type: ActionType.BookmarkPost,
+        step: {
+          title: 'Bookmark your first post',
+          description: 'Save posts you find valuable for easy access later.',
+        },
+        actions,
+      }),
+      createChecklistStep({
+        type: ActionType.DigestConfig,
+        step: {
+          title: 'Configure your personalized digest',
+          description:
+            'Adjust the frequency and send time of your personalized digest to help build a habit and stay updated.',
+          component: (props) => {
+            return (
+              <ChecklistStep {...props}>
+                <div className="flex">
+                  <Button
+                    tag="a"
+                    variant={ButtonVariant.Primary}
+                    size={ButtonSize.XSmall}
+                    href={`${webappUrl}account/notifications`}
+                    onClick={() => {
+                      completeAction(ActionType.DigestConfig);
+                    }}
+                  >
+                    Set up now
+                  </Button>
+                </div>
+              </ChecklistStep>
+            );
+          },
+        },
+        actions,
+      }),
+      createChecklistStep({
+        type: ActionType.StreakMilestone,
+        step: {
+          title: 'Reach the first streaks milestone',
+          description:
+            'Read a post for two days in a row (or more) to achieve your first reading streak milestone.',
+          component: (props) => {
+            return (
+              <ChecklistStep {...props}>
+                <div className="flex">
+                  <Button
+                    variant={ButtonVariant.Primary}
+                    size={ButtonSize.XSmall}
+                    onClick={() => {
+                      const readingStreakHeaderButton = document.getElementById(
+                        'reading-streak-header-button',
+                      );
+
+                      if (readingStreakHeaderButton) {
+                        readingStreakHeaderButton.click();
+                      }
+                    }}
+                  >
+                    Track your streak
+                  </Button>
+                </div>
+              </ChecklistStep>
+            );
+          },
+        },
+        actions,
+      }),
+    ];
+  }, [isChecklistReady, actions, openModal, completeAction]);
+
+  const checklist = useChecklist({ steps });
+
+  const [checklistView, setChecklistView] =
+    usePersistentContext<ChecklistViewState>(
+      generateStorageKey(StorageTopic.Onboarding, 'sidebarCard', user?.id),
+      ChecklistViewState.Open,
+    );
```

Comment:
Will be replace with settings when https://github.com/dailydotdev/daily-api is merged

Reply:
So how will we determine what takes priority the setting or the experiment?

Reply:
Ah actually this reminded me also that we need to mess with the sidebarExpanded as well for this experiment right?

Reply:
See comments in https://github.com/dailydotdev/daily-api/pull/1983#issuecomment-2162449306

Reply:
Yeah, asking just because I don't see those changes yet here. They will be in this PR right?

Reply:
Yes, was waiting for settings API side merged.

Reply:
0dc6ed3f3121b716c5f44f4629eb79b6da91c13d